openshift_day2_configurator/configurators/ingress.py

```diff
@@ -0,0 +1,377 @@
+
+    except Exception as ex:
+        logger.error(f"Failed to create {new_ingress_certificate_filename} Ingress certificate file: {ex}")
+        return {CREATE_NEW_INGRESS_CERTIFICATE: {"res": False, "err": str(ex)}}
+
+    return {CREATE_NEW_INGRESS_CERTIFICATE: {"res": True, "err": ""}}
+
+
+def create_ingress_certificate_configmap(
+    client: DynamicClient,
+    logger: logging.Logger,
+    cluster_domain: str,
+) -> Dict[str, Dict[str, str]]:
+    logger.debug(CREATE_INGRESS_CERTIFICATE_CONFIGMAP)
+
+    ingress_certificate = get_ingress_certificate_from_file(cluster_domain=cluster_domain)
+
+    return {
+        CREATE_INGRESS_CERTIFICATE_CONFIGMAP: create_ocp_resource(
+            ocp_resource=ConfigMap(
+                client=client,
+                name=INGRESS_CERTIFICATE_CONFIGMAP_NAME,
+                namespace="openshift-config",
+                data={
+                    "ca-bundle.crt": ingress_certificate,
+                },
+            ),
+            logger=logger,
+        )
+    }
+
+
+def update_cluster_proxy_trusted_ca(
+    client: DynamicClient,
+    logger: logging.Logger,
+) -> Dict[str, Dict[str, Union[str, bool]]]:
+    logger.debug(UPDATE_CLUSTER_PROXY_TRUSTED_CA)
+
+    cluster_proxy = Proxy(client=client, name="cluster")
+
+    if not cluster_proxy.exists:
+        cluster_proxy_not_exist_err: str = f"Cluster Proxy {cluster_proxy.name} does not exist"
+        logger.error(cluster_proxy_not_exist_err)
+        return {
+            UPDATE_CLUSTER_PROXY_TRUSTED_CA: {
+                "res": False,
+                "err": cluster_proxy_not_exist_err,
+            }
+        }
+
+    proxy_dict = {
+        "trustedCA": {
+            "name": INGRESS_CERTIFICATE_CONFIGMAP_NAME,
+        }
+    }
+
+    try:
+        ResourceEditor({cluster_proxy: {"spec": proxy_dict}}).update()
+    except Exception as ex:
+        logger.debug(f"Failed to update cluster proxy with error {ex}")
+        return {UPDATE_CLUSTER_PROXY_TRUSTED_CA: {"res": False, "err": str(ex)}}
+
+    return {UPDATE_CLUSTER_PROXY_TRUSTED_CA: {"res": True, "err": ""}}
+
+
+def create_wildcard_certificate_tls_secret(
+    client: DynamicClient,
+    logger: logging.Logger,
+    cluster_domain: str,
+) -> Dict[str, Dict[str, str]]:
+    logger.debug(CREATE_WILDCARD_CERTIFICATE_TLS_SECRET)
+
+    ingress_certificate = get_ingress_certificate_from_file(cluster_domain=cluster_domain)
+
+    with open(f"{cluster_domain}.key", "rb") as ingress_certificate_key_file:
+        ingress_certificate_key = ingress_certificate_key_file.read().decode("utf-8")
+
+    return {
+        CREATE_WILDCARD_CERTIFICATE_TLS_SECRET: create_ocp_resource(
+            ocp_resource=Secret(
+                client=client,
+                name=f"wildcard.{cluster_domain}",
+                namespace=OPENSHIFT_INGRESS_NAMESPACE,
+                data_dict={
+                    "tls.crt": base64.b64encode(ingress_certificate.encode("utf-8")).decode("utf-8"),
```

Comment:
please create a util function for encodfing

Reply:
done